config/routes.rb

```diff
@@ -62,7 +62,8 @@
       resources :verify_account, only: :create do
         post '/activate', to: 'verify_account#activate', on: :collection
       end
-      resources :site_settings, only: :show, param: :name
+      resources :site_settings, only: %i[show index], param: :name
```

Comment:
The `#update` API itself should be in a controller under the `Admin` namespace IMO, so that alone will solve your issue.
Otherwise for similar cases it's better to have a unified `param` so having the `:id` here may not be a good idea (more on that given above.).